src/INode.java

```diff
@@ -189,9 +189,29 @@ private ASTValue evalAnd(WordsEnvironment environment) {
 		throw new AssertionError("Not yet implemented");
 	}
 
-	private ASTValue evalAssign(WordsEnvironment environment) {
-		// TODO
-		throw new AssertionError("Not yet implemented");	
+	private ASTValue evalAssign(WordsEnvironment environment) throws WordsRuntimeException {
+		WordsObject obj = children.get(0).eval(environment).objValue;
```

Comment:
Actually, per the grammar, the first (index zero) child is a REFERENCE_LIST node, and the evaluation of this node should indeed return an object (not a string).  So I think the existing code is correct.